### PR TITLE
fix redboxes on mobile. make mockstore work better

### PR DESCRIPTION
**Pull request by chrisnojima**
Fixes redbox issues on mobile.

- [x] make sure visdiff on desktop looks ok

### CODE REVIEW

File: shared/chat/conversation/messages/text/container.js [hunk 1]
```diff
@@ -11,10 +11,10 @@ import type {OwnProps} from './container'
 
 const getProps = createCachedSelector(
   [Constants.getMessageFromMessageKey, Constants.getEditingMessage],
-  (message: Constants.TextMessage, editingMessage) => ({
+  (message: ?Constants.TextMessage, editingMessage) => ({
     isEditing: message === editingMessage,
-    text: message.message.stringValue(),
-    type: message.messageState,
+    text: message ? message.message.stringValue() : null,
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/text/container.js]: its technically possible the store won't have a value here, happens in mock but never in production hopefully

File: shared/chat/conversation/rekey/participant-rekey.native.js [hunk 2]
```diff
@@ -61,7 +61,6 @@ const containerStyle = {
   ...globalStyles.flexBoxColumn,
   alignItems: 'stretch',
   backgroundColor: globalColors.darkBlue4,
-  borderLeft: `1px solid ${globalColors.black_20}`,
```
Comment (chrisnojima) [inline shared/chat/conversation/rekey/participant-rekey.native.js]: no borderLeft on native

File: shared/common-adapters/autosize-input.native.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 // @flow
 
 // FIXME: stub placeholder for mobile version
-function AutosizeInput() {}
+function AutosizeInput() {
+  return null
```
Comment (chrisnojima) [inline shared/common-adapters/autosize-input.native.js]: we have to return null, not undefined

File: shared/common-adapters/markdown.shared.js [hunk 4]
```diff
@@ -30,9 +30,9 @@ function processAST(ast, createComponent) {
   return ast.component
 }
 
-export function parseMarkdown(markdown: string = '', markdownCreateComponent: MarkdownCreateComponent) {
+export function parseMarkdown(markdown: ?string, markdownCreateComponent: MarkdownCreateComponent) {
```
Comment (chrisnojima) [inline shared/common-adapters/markdown.shared.js]: this only overrides if you pass undefined, not null

File: shared/dev/dumb-sheet/render.native.js [hunk 5]
```diff
@@ -22,6 +22,7 @@ class DumbSheetRender extends Component<void, Props, any> {
     this.state = {
       filterShow: false,
       localFilter: props.dumbFilter || '',
+      mockStore: null,
```
Comment (chrisnojima) [inline shared/dev/dumb-sheet/render.native.js]: use mockStore in state to force a re-render

File: shared/dev/dumb-sheet/render.native.js [hunk 6]
```diff
@@ -131,26 +132,32 @@ class DumbSheetRender extends Component<void, Props, any> {
...
+
     console.log('test render idx', this.state.testIndex, component)
     return <Box>{this._makeStoreWrapper(component)}</Box>
   }
 
   _updateMockStore(mockStore: any) {
-    if (mockStore) {
-      if (!this._mockStore) {
-        this._mockStore = createStore(old => mockStore, mockStore)
+    if (mockStore === this._mockStore) return true
```
Comment (chrisnojima) [inline shared/dev/dumb-sheet/render.native.js]: it's the same, so its safe to continue to render

File: shared/dev/dumb-sheet/render.native.js [hunk 7]
```diff
@@ -131,26 +132,32 @@ class DumbSheetRender extends Component<void, Props, any> {
...
+
+    setImmediate(() => {
```
Comment (chrisnojima) [inline shared/dev/dumb-sheet/render.native.js]: calling replaceReducer will actually fire off connectors which causes react to be very unhappy if we're mounting / unmounting. instead we call setState and manage this slightly more

File: shared/login/forms/index.native.js [hunk 8]
```diff
@@ -154,7 +154,6 @@ const stylesBannerBlue = {
   paddingLeft: globalMargins.medium,
   paddingRight: globalMargins.medium,
   paddingTop: globalMargins.tiny,
-  textAlign: 'center',
```
Comment (chrisnojima) [inline shared/login/forms/index.native.js]: no textAlign on boxes

**Comment by chrisnojima** [comment]
@keybase/react-hackers this fixes dumb sheets on mobile

**Comment by keybase-ci-visdiff** [comment]
The commits 970a2d4d6a53...2ba0c3285b4a introduce visual changes on linux. <details><summary>:mag_right: 6 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Tracker-New_user_Scroll_1-970a2d4d6a53.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Tracker-New_user_Scroll_1-2ba0c3285b4a.png)  
   Tracker-New_user_Scroll_1 comparison
 * Changed: **Error_View-User_Not_Found** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-User_Not_Found-970a2d4d6a53.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-User_Not_Found-2ba0c3285b4a.png)  
   Error_View-User_Not_Found comparison
 * Changed: **Error_View-Key_Not_Found** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Key_Not_Found-970a2d4d6a53.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Key_Not_Found-2ba0c3285b4a.png)  
   Error_View-Key_Not_Found comparison
 * Changed: **Error_View-Key_No_Secret** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Key_No_Secret-970a2d4d6a53.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Key_No_Secret-2ba0c3285b4a.png)  
   Error_View-Key_No_Secret comparison
 * Changed: **Error_View-Device_No_Provision** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Device_No_Provision-970a2d4d6a53.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Device_No_Provision-2ba0c3285b4a.png)  
   Error_View-Device_No_Provision comparison
 * Changed: **Error_View-Bad_Login_Password** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Bad_Login_Password-970a2d4d6a53.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496856686784-970a2d4d6a53-2ba0c3285b4a-linux/Error_View-Bad_Login_Password-2ba0c3285b4a.png)  
   Error_View-Bad_Login_Password comparison
</summary>

**Comment by chrisnojima** [comment]
visdiff failed, doing a rebuild

**Comment by keybase-ci-visdiff** [comment]
The commits 8952c6de34ca...a749ae74e9eb introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Error_View-User_Not_Found** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-User_Not_Found-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-User_Not_Found-a749ae74e9eb.png)  
   Error_View-User_Not_Found comparison
 * Changed: **Error_View-Key_Not_Found** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-Key_Not_Found-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-Key_Not_Found-a749ae74e9eb.png)  
   Error_View-Key_Not_Found comparison
 * Changed: **Error_View-Key_No_Secret** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-Key_No_Secret-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-Key_No_Secret-a749ae74e9eb.png)  
   Error_View-Key_No_Secret comparison
 * Changed: **Error_View-Device_No_Provision** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-Device_No_Provision-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497365047795-8952c6de34ca-a749ae74e9eb-linux/Error_View-Device_No_Provision-a749ae74e9eb.png)  
   Error_View-Device_No_Provision comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 8952c6de34ca...82f3d3807911 introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Error_View-User_Not_Found** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-User_Not_Found-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-User_Not_Found-82f3d3807911.png)  
   Error_View-User_Not_Found comparison
 * Changed: **Error_View-Key_Not_Found** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-Key_Not_Found-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-Key_Not_Found-82f3d3807911.png)  
   Error_View-Key_Not_Found comparison
 * Changed: **Error_View-Key_No_Secret** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-Key_No_Secret-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-Key_No_Secret-82f3d3807911.png)  
   Error_View-Key_No_Secret comparison
 * Changed: **Error_View-Device_No_Provision** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-Device_No_Provision-8952c6de34ca.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497370669258-8952c6de34ca-82f3d3807911-linux/Error_View-Device_No_Provision-82f3d3807911.png)  
   Error_View-Device_No_Provision comparison
</summary>

**Pull request merged**